### PR TITLE
UIManager API

### PR DESCRIPTION
**Pull request by sgny**
Relates to #483 

This is an incomplete implementation of methods in `UIManager` but enabling `LayoutAnimation` and showing and dismissing popups on Android are included.

`setJSResponder` and `clearJSResponder` are for finer grained control of which component should be the gesture responder (such as for creating a swipable element in a `ScrollView`).

Some of the other methods have already been covered elsewhere - such as `measure` in `NativeMethods` and therefore omitted, but I will revisit this if I'm able to document other methods.

### CODE REVIEW

File: reason-react-native/src/apis/UIManager.rei [hunk 1]
```diff
@@ -0,0 +1,29 @@
+type popupAction;
```
Comment (cknitt) [inline reason-react-native/src/apis/UIManager.rei]: If we make this an abstract type, I think we should also expose the corresponding constants (UIManagerModuleConstants.ACTION_ITEM_SELECTED and UIManagerModuleConstants.ACTION_DISMISSED on the Java side). Not sure if/where they are exported to JS though.

Comment (sgny) [reply reason-react-native/src/apis/UIManager.rei]: They were exposed as `PopupMenu= { dismissed: 'dismissed', itemSelected: 'itemSelected' }`.

**Comment by sgny** [comment]
> 👍

Thanks. I'll postpone merging to see if you or @MoOx have further comments.

**Comment by sgny** [comment]
> Did refmt allowed this long lines?

No 😞

**Pull request merged**